docs/blueprints/001-installation-of-extensions.md

```diff
@@ -0,0 +1,114 @@
+
+* installer command line arguments that the user can supply to indicate which optional extensions need to be installed (e.g. fuseml-install --extension mlflow --extension kfserving)
+* make it possible to install/uninstall extensions in follow-up installer invocations, separately from the main installation
+* MLFlow no longer installed by default
+* add support to install KFServing and KNative.
+
+## Current state
+
+`fuseml-installer` has two ways of installing components
+
+* `fuseml-installer install` - installs all necessary components (but actually _also_ MLFlow, which should be an extension)
+* `make` targets. Specific targets defined in the `Makefile` describe the installation of other components, like KFServing
+
+## Proposed solution
+
+### Makefile
+
+Get rid of specific installation targets that install something for production usage (e.g. not used by testing/development).
+
+### CLI
+
+1. Implement new option for `install` command
+
+  `fuseml-installer install [--extension <name>] [--extension-repository <path>]`
+
+  This command installs all necessary components AND extensions selected by `--extension` option. Optionally uses path specified
+  by `--extension-repository` option as a source for extensions that should be installed. Path could be local directory on URL.
+
+
+2. Implement new command for managing extensions. This could be useful for adding an extension to already configured system.
+
+  `fuseml-installer extension`
+
+  This should have several sub commands, like
+
+  * `fuseml-installer extension list` - list installed extensions
+  * `fuseml-installer extension available [--extension-repository <path>]` = list extensions available for installation
+  * `fuseml-installer extension add <name> [--extension-repository <path>]` - install selected extension
+  * `fuseml-installer extension remove <name> [--extension-repository <path>]` - ubinstalls selected extension
+
+
+### Description files
+
+Extension repository is a directory (local or remote) containing description files of extensions. Each description file provides 
+description of operations that could be done with the extension.
+
+Each extension has its own subdirectory that matches the extension name. Mandatory component under each extension subdirectory is the
+description file, in `yaml` format.
+
+Directory structure:
+
+```
+extensions
+ - installation
+    - mlflow
+      - description.yaml
+    - knative
+      - description.yaml
+      - install.sh
+      - uninstall.sh
+```
+
+Possible extension types are
+
+* `helm` - helm chart. For helm chart type, no extra install/uninstall commands are necessary
+* `manifest` - Kubernetes manifest, to be installaded using `kubectl`. No extra install/uninstall commands are necessary. All information (including namespace for example) are expected to be present in the manifest file.
+* `script` - shell script. Specific `install` and `uninstall` actions need to be provided by way of referencing specific scripts.
```

Comment:
`script` here looks scary, right? But currently we can't get past it, see e.g. how we install knative https://github.com/fuseml/fuseml/blob/main/scripts/knative-install.sh

Reply:
We  might agree to have `script` typo for now (so we have working solutions for extension installation soon) and aim for replacing existing scripts with manifests or helm charts.... 

We might actually create our own helm charts, when they are missing.

Reply:
I suggested a hybrid approach in another comment: using a list of installation steps provides us with a smooth transition from one `script` step to several `helm` and `manifest` steps.

Reply:
Yes, this would definitely help e.g. with seldon-core installation https://github.com/fuseml/fuseml/blob/main/scripts/seldon-operator-install.sh

However not necessary with the others, see e.g. the script for istio installation https://github.com/fuseml/fuseml/blob/main/scripts/istio-minimal-install.sh
(and no, I do not plan to move istio to the extension, just pointing out the example where script seems to be necessary for the time being)

Reply:
Updated with latest commit